src/sql/workbench/services/objectExplorer/browser/asyncServerTreeDataSource.ts

```diff
@@ -51,17 +51,14 @@ export class AsyncServerTreeDataSource implements IAsyncDataSource<ConnectionPro
 			} else if (element instanceof ConnectionProfileGroup) {
 				return element.getChildren();
 			} else if (element instanceof TreeNode) {
-				if (element.children) {
-					return element.children;
-				} else {
-					return await this._objectExplorerService.resolveTreeNodeChildren(element.getSession()!, element);
-				}
+				return await this._objectExplorerService.resolveTreeNodeChildren(element.getSession()!, element);
 			}
 		} catch (err) {
 			if (element instanceof TreeNode) {
 				element.errorStateMessage = err.message ?? err;
 			}
-			if (err.message) {
+			// In case of connection profile, we won't show the error here and let the connection service handle it.
+			if (err.message && !(err instanceof ConnectionError)) {
```

Comment:
err could be anything, not necessarily an Error object. I'd suggest using [getErrorMessage](https://github.com/Microsoft/azuredatastudio/blob/main/src/vs/base/common/errors.ts#L202) and then using that in all the places you want to get the error message in this catch.